gen_pools_and_gauges.py

```diff
@@ -119,19 +116,22 @@ def main():
         gauge_info = BalPoolsGauges(chain)
         # pools
         # TODO: consider moving to query object??
-        result = process_query_swap_enabled_pools(query_swap_enabled_pools(chain))
-        if result:
-            pools[chain] = result
-
-        # gauges
-        if chain in NO_GAUGE_SUBGRAPH:
-            # no (gauge) subgraph exists for this chain
+        try:
+            result = process_query_swap_enabled_pools(query_swap_enabled_pools(chain))
+            if result:
+                pools[chain] = result
+        except Exception as e:
+            print(f"Error using core subgraph for {chain}, skipping: {e}")
             continue
-        result = process_query_preferential_gauges(
-            gauge_info.query_preferential_gauges()
-        )
-        if result:
-            gauges[chain] = result
+        # gauges
+        try:
+            result = process_query_preferential_gauges(
+                gauge_info.query_preferential_gauges()
+            )
+            if result:
+                gauges[chain] = result
+        except Exception as e:
+            print(f"Error using gauge subgraph for {chain}, skipping: {e}")
```

Comment:
also here probably better to initialise `gauges[chain] = {}`

Reply:
Updated